### DISCUSSION TITLE
Testing your plugin?

### DISCUSSION THREAD

**Discussion by lufinkey** [SteamDeckHomebrew/decky-loader#298] [General]
I've been trying to set up a dev environment to make a decky plugin and have been running into some roadblocks:

- I can't seem to find any guide for building the full plugin itself to install in decky. `pnpm run build` builds the frontend code, but generating the full install-ready plugin seems to be a challenge. I've tried directly installing the `dist` folder, the repo folder, the zip that github provides, etc, but nothing seemed to work.
- can't seem to test decky in a non-steamdeck environment. The steam deck UI can be started on any linux machine with steam installed through the command `steam -gamepadui`. However, the "Developer Mode" switch isn't available through this (obviously not the fault of this plugin, but can't seem to find any workaround or guide for developing)

Could someone illuminate this for me? If I end up figuring it out myself I'd be happy to add it as a PR to the website or README, but I can't seem to find anything on this.

**Comment by PartyWumpus** [comment]
You can enable CEF debugging without the dev menu by adding an empty file called `.cef-enable-remote-debugging` to `~/.steam/steam/` (this will probably be put into the installer soon)

**Reply by lufinkey** [reply]
Does this enable debugging only for localhost, or for anybody on the network?

**Reply by PartyWumpus** [reply]
normally it would enable it for the network, but decky disables the forwarding so it's only on localhost.
decky has a toggle to re-enable forwarding if you want it.
now that i think about it, it might be a good idea to disable CEF debugging when decky is uninstalled...

**Comment by jurassicplayer** [comment]
For deploying a test version to your console, there are vscode tasks in the [decky-plugin-template repo](https://github.com/SteamDeckHomebrew/decky-plugin-template) that you can use/look at for how to build/deploy onto the console via ssh. You will need to make a copy of `.vscode/defsettings.json` as `.vscode/settings.json` with your deck's configurations if you use the vscode tasks.

For creating a release .zip I have some extra [vscode tasks](https://github.com/jurassicplayer/decky-autoflatpaks/blob/main/.vscode/tasks.json) that you can use as reference (or copy).

**Comment by gustakasn0v** [comment]
This was super helpful, thanks for sharing! I'll make a PR to https://github.com/SteamDeckHomebrew/wiki to document this in https://wiki.deckbrew.xyz/en/plugin-dev/getting-started as I struggled with the same issue and couldn't find guidance on how to set my dev environment up (**edit**: just noticed @lufinkey wanted to do a PR as well, do you want to collab on this or happy for me to take the first stab? Asking because documenting my setup while I work on https://github.com/saumya-banthia/tailscale-control/pull/5 will be so much easier)

**Comment by gustakasn0v** [comment]
Sidenote: does anybody know how to test the React side of a plugin locally? Thinking something along the lines of the typical [webpack dev server](https://webpack.js.org/configuration/dev-server) that most React setups have. 

In the meantime I'll take a stab at setting it up myself and share my findings

**Reply by devnoot** [reply]
Wondering the same thing. Did you ever find an answer?
